packages/core/streams/ai-stream.ts

```diff
@@ -5,110 +5,142 @@ import {
   type ReconnectInterval
 } from 'eventsource-parser'
 
+/**
+ * Interface representing callbacks for an AIStream.
+ * @interface
+ */
 export interface AIStreamCallbacks {
   onStart?: () => Promise<void>
   onCompletion?: (completion: string) => Promise<void>
   onToken?: (token: string) => Promise<void>
 }
 
+/**
+ * Interface representing a custom parser for AIStream data.
+ * @interface
+ */
 export interface AIStreamParser {
   (data: string): string | void
 }
 
-export function createEventStreamTransformer(customParser: AIStreamParser) {
-  const decoder = new TextDecoder()
-  let parser: EventSourceParser
+/**
+ * Creates a transform stream that parses events from an EventSource stream.
+ * @param {AIStreamParser} customParser - A custom parser function.
+ * @return {TransformStream<Uint8Array, string>} A transform stream.
+ */
+export function createEventStreamTransformer(
+  customParser: AIStreamParser
+): TransformStream<Uint8Array, string> {
+  const textDecoder = new TextDecoder()
+  let eventSourceParser: EventSourceParser
 
-  return new TransformStream<Uint8Array, string>({
+  return new TransformStream({
     async start(controller): Promise<void> {
-      function onParse(event: ParsedEvent | ReconnectInterval) {
-        if (event.type === 'event') {
-          const data = event.data
-          if (data === '[DONE]') {
+      eventSourceParser = createParser(
+        (event: ParsedEvent | ReconnectInterval) => {
+          if (
+            'data' in event &&
+            event.type === 'event' &&
+            event.data === '[DONE]'
+          ) {
             controller.terminate()
             return
           }
 
-          const message = customParser(data)
-          if (message) controller.enqueue(message)
+          if ('data' in event) {
+            const parsedMessage = customParser(event.data)
+            if (parsedMessage) controller.enqueue(parsedMessage)
+          }
         }
-      }
-
-      parser = createParser(onParse)
+      )
     },
 
     transform(chunk) {
-      parser.feed(decoder.decode(chunk))
+      eventSourceParser.feed(textDecoder.decode(chunk))
     }
   })
 }
 
 /**
- * This stream forks input stream, allowing us to use the result as a
```

Comment:
This is a more informative comment. Lets not lose important contributor information that could be helpful for newcomers.